src/main/java/org/broadinstitute/hellbender/utils/read/markduplicates/MarkDuplicatesScoringStrategy.java

```diff
@@ -25,10 +26,13 @@
 
     /**
      * Given a read, compute the reads's score for mark duplicates.
+     *
+     * NOTE: this code is intended to match {@link htsjdk.samtools.DuplicateScoringStrategy.ScoringStrategy} behavior.
      */
-    public int score(final GATKRead read){
+    public short score(final GATKRead read){
         Utils.nonNull(read);
-        return scoring.applyAsInt(read);
+        // We max out at Short.MAX_VALUE / 2 to prevent overflow if we add two very high quality/length reads into pairs
```

Comment:
the negative value if it fails vendor quality is strange, maybe mention that in a comment?